packages/hardhat/contracts/strategy/CrabStrategyV2.sol

```diff
@@ -181,11 +192,44 @@ contract CrabStrategyV2 is StrategyBase, StrategyFlashSwap, ReentrancyGuard, Own
         require(msg.sender == weth || msg.sender == address(powerTokenController), "Cannot receive eth");
     }
 
+    /**
+     * @notice initializes the collateral ratio upon the first migration
+     */
+    function initialize(
+        uint256 wSqueethToMint,
+        uint256 crabSharesToMint,
+        uint256 timeAtLastHedge,
+        uint256 priceAtLastHedge
+    ) external payable {
+        require(msg.sender == crabMigration, "not Crab Migration contract");
+        require(!isInitialized, "Crab V2 already initialized");
+
+        uint256 amount = msg.value;
+        uint256 strategyDebt;
+        uint256 strategyCollateral;
+
+        _checkStrategyCap(amount, strategyCollateral);
+
+        require((strategyDebt == 0 && strategyCollateral == 0), "C5");
+        // store hedge data as strategy is delta neutral at this point
+        // only execute this upon first deposit
+        uint256 wSqueethEthPrice = IOracle(oracle).getTwap(ethWSqueethPool, wPowerPerp, weth, hedgingTwapPeriod, true);
+        timeAtLastHedge = block.timestamp;
```

Comment:
timeAtLastHedge and priceAtLastHedge shadowed inputs

Reply:
What do you mean shadow inputs?

Reply:
they're coming in as params on 201 and 202 then gettting set on 217 and 218 so the inputs don't do anything